python/pyspark/pandas/frame.py

```diff
@@ -6826,8 +6826,28 @@ def drop(
                     )
             return DataFrame(internal)
 
+    def _prepare_sort_by_scols(self, by: Union[Name, List[Name]]) -> List[Column]:
+        if is_name_like_value(by):
```

Comment:
Extract `_prepare_sort_by_scols` in order to make `nlargest/nsmallest` rely on `_sort` rather than `sort_values`, since `sort_values` has no `keep` parameter.